src/main/scala/temple/generate/kube/KubernetesGenerator.scala

```diff
@@ -39,8 +40,46 @@ object KubernetesGenerator {
   private def generateDbService(service: Service): String =
     generateHeader(service, GenType.Service, isDb = true)
 
-  private def generateDbDeployment(service: Service): String =
-    generateHeader(service, GenType.Deployment, isDb = true)
+  private def generateDbDeployment(service: Service): String = {
+    val name = service.name + "-db"
+    val deploymentBody = Body(
+      DeploymentSpec(
+        1,
+        Selector(Labels(service.name, GenType.Deployment, isDb = true)),
+        strategy = Some(Strategy("Recreate")),
+        Template(
+          Metadata(name, Labels(service.name, GenType.Deployment, isDb = true)),
+          PodSpec(
+            name,
+            Seq(
+              Container(
+                service.dbImage,
+                name,
+                ports = Seq(),
+                env = service.envVars.map(x => EnvVar(x._1, x._2)),
+                volumeMounts = Seq(
+                  VolumeMount("/var/lib/postgresql/data", None, name + "-claim"),
+                  VolumeMount("/docker-entrypoint-initdb.d/init.sql", Some("init.sql"), name + "-init"),
```

Comment:
These are making assumptions about the selected database: you should be taking that as an argument and switching on it